instruqt-tracks/terraform-basics-aws/track.yml

```diff
@@ -0,0 +1,472 @@
+
+      Check out this tutorial for step-by-step instructions:
+
+      https://learn.hashicorp.com/terraform/getting-started/install.html
+
+      We've pre-installed Terraform in your Instruqt lab environment for you.
+  tabs:
+  - title: Shell
+    type: terminal
+    hostname: workstation
+  - title: Code Editor
+    type: service
+    hostname: workstation
+    port: 8080
+  difficulty: basic
+  timelimit: 300
+- slug: aws-credentials
+  id: 6yijz7bt56nc
+  type: challenge
+  title: "\U0001F510 Connecting Terraform to AWS"
+  teaser: Connecting to AWS with Access Keys and Secret Keys.
+  assignment: |-
+    **Goal**: Understand methods for configuring AWS provider credentials.
+    **Estimated time**: 3-5 minutes.
+
+    In order to authenticate to AWS and build resources, Terraform requires you to provide a set of credntials, backed by an appropriate IAM policy.
+
+    These are part of the provider configuration, and could be set within the provider stanza.
+
+    ```
+    provider "aws" {
+      region  = "us-east-1"
+      access_key = "my-access-key"
+      secret_key = "my-secret-key"
+    }
+    ```
+
+    Warning: Hard-coding credentials into any Terraform configuration is not recommended, and risks secret leakage should this file ever be committed to a public version control system.
+
+    You can also use an AWS credentials file to specify your credentials. The default location is $HOME/.aws/credentials on Linux and macOS, or "%USERPROFILE%\.aws\credentials" for Windows users.
+
+    For this training environment, we have prepared some temporary AWS credentials and stored them as environment variables. Terraform will automatically read and use the environment variables that are configured in your shell environment.
+
+    1. Run the following commands in the terminal to see your temporary credentials:
+
+    ```
+    env | grep AWS_ACCESS_KEY_ID
+    env | grep AWS_SECRET_ACCESS_KEY
+    ```
+  notes:
+  - type: text
+    contents: Did you know HCL stands for "Hashicorp COnfiguration Language"?
+  tabs:
+  - title: Code Editor
+    type: service
+    hostname: workstation
+    port: 8080
+  - title: Shell
+    type: terminal
+    hostname: workstation
+  difficulty: basic
+  timelimit: 300
+- slug: terraform-code
+  id: r2xd5ubxvdnh
+  type: challenge
+  title: "\U0001F468‍\U0001F4BB What does Terraform code look like?"
+  teaser: The terraform DSL (Domain Specific Language) is a declarative language that
+    lets you build almost any type of infrastructure.
+  assignment: |-
+    We've downloaded some terraform code onto your workstation.
+
+    1. Run the following command in the terminal to see the terraform files in your working directory:
+
+    ```
+    ls *.tf
+    ```
+
+    2. Switch tabs to VSCode. You will note the same files are visible in the file explorer pane on the left. Terraform files are marked with the purple T icon.
```

Comment:
The student already begins the lab in VSCode.  The Shell tab should be placed first since the first thing we ask the student to do is run a command.